src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -1334,6 +1336,95 @@ select Task.Factory.StartNew(() => pair.Item1.Receive(new byte[1]), Cancellation
         }
     }
 
+    public class BlockingSendReceive
+    {
+
+        private readonly ITestOutputHelper _log;
+        public static readonly object[][] Loopbacks = MemberDatas.Loopbacks;
+
+        public BlockingSendReceive(ITestOutputHelper output)
+        {
+            _log = output;
+        }
+
+        [Theory]
+        [MemberData(nameof(Loopbacks))]
+        public async Task BlockingStreamRead_Close(IPAddress ipAddress)
+        {
+            // Test for #22564 when close is called while blocking read is pending.
+            using (var server = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
+            using (var client = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
+            {
+                server.BindToAnonymousPort(ipAddress);
+                server.Listen(1);
+                byte[] buffer = new byte[1];
+                buffer[0] = Convert.ToByte('a');
+
+                Task clientTask = Task.Run(() =>
+                {
+                    client.Connect(server.LocalEndPoint);
+                    client.Send(buffer);
+                    // Blocking read.
+                    try
+                    {
+                        client.Receive(buffer);
+                    }
+                    catch (SocketException) { }
+                    catch (ObjectDisposedException) { }
+                });
+                Socket s = server.Accept();
+                // Try to receive something so we know client thread is running.
+                s.Receive(buffer, 1, SocketFlags.None);
+                // There is however race condition that if the Receive() is not started yet, this will simply
+                // finish without hitting the test scenario.
+                await Task.Delay(500);
+
+                // clientThread should be now blocked on Receive()
+                // Close client socket from parent thread.
+                client.Close();
+                clientTask.Wait(TestSettings.PassingTestTimeout);
+            }
+        }
+
+        [Theory]
+        [PlatformSpecific(~TestPlatforms.OSX)]
```

Comment:
Is there a separate issue for OSX? Can we link the issue here?

Reply:
not at the moment, both issues were reported on Linux.  I can update #26034 and leave it open for OSX instead of closing.